nixos/modules/services/networking/iwd.nix

```diff
@@ -22,6 +147,17 @@ in {
 
     systemd.packages = [ pkgs.iwd ];
 
+    systemd.services.iwd.preStart = mkIf (cfg.networks != {}) (concatStrings (
+      flip mapAttrsToList predefinedNetworks (file: content: ''
+        ln -sf '${content.file}' "${encodeIfNeeded file content.extension}"
```

Comment:
Maybe dumb question, but does the preStart run outside of whatever systemd bits may be limiting iwd's filesystem accesses at runtime? I'm not sure which directives are enabled anyway, but curious as offhand I don't know if that's already locked down in preStart or not.  I suppose if it works at all then that means it's good :).

Reply:
At least not yet, I'm using this change for several months and the preStart hook works fine. If the upstream systemd unit (we're currently using and modifying in the module) becomes more strict, we'd have to fix it accordingly IMHO.